src/nfa/operations.cc

```diff
@@ -927,29 +927,72 @@ Nfa mata::nfa::algorithms::minimize_brzozowski(const Nfa& aut) {
     return determinize(revert(determinize(revert(aut))));
 }
 
-Nfa mata::nfa::minimize(
-                const Nfa& aut,
-                const ParameterMap& params)
-{
-    Nfa result;
-    // setting the default algorithm
-    decltype(algorithms::minimize_brzozowski)* algo = algorithms::minimize_brzozowski;
+Nfa mata::nfa::minimize_nfa(const Nfa& aut, const ParameterMap& params) {
     if (!haskey(params, "algorithm")) {
         throw std::runtime_error(std::to_string(__func__) +
             " requires setting the \"algorithm\" key in the \"params\" argument; "
             "received: " + std::to_string(params));
     }
 
+    // Setting the algorithm. Default is Hopcroft.
     const std::string& str_algo = params.at("algorithm");
-    if ("brzozowski" == str_algo) {  /* default */ }
+    decltype(algorithms::minimize_brzozowski)* algo = algorithms::minimize_brzozowski;
+    if (str_algo == "brzozowski") { /* default */ }
     else {
+        throw std::runtime_error(std::string(__func__) +
+            " received an unknown value for the \"algorithm\" key: " + str_algo);
```

Comment:
Shouldn't we check for hopcroft here and call determinization + hopcroft?

Reply:
I'm not sure if we should allow the use of algorithms for DFAs on NFAs.